migration/src/main/resources/migrations/ensure-orders-have-encounters.xml

```diff
@@ -17,28 +17,43 @@
            then create the encounters for orders that don’t have encounters
         -->
         <sql>
-            There are over 300,000 orders on both Rwink and Butaro with no linked encounter
-            select count(*) from orders where encounter_id is null
+            SET @EncTUUID :=  "a9ef19d1-8875-11ea-8c74-7a7919290ad6";
 
-            Of these, there are still between 100k and 150k on each server with no encounters on the same day
 
-            select count(*)
-            from orders o
-            left join encounter e on o.patient_id = e.patient_id and date(o.start_date) = date(e.encounter_datetime)
-            where e.encounter_id is null;
+            /* Create EncounterType */
+            INSERT INTO `encounter_type`
+            (`name`,`description`,`creator`,`date_created`,`uuid`)
+            VALUES
+            (
+            "Drug Order Encounter","for orders without encounters",1,now(),@EncTUUID);
 
-            So we need to figure out a strategy for this.
+            /* Get encounterType id */
+            select @encTId:=encounter_type_id from encounter_type where uuid=@EncTUUID;
 
-            Safest thing for now might be to create new encounters for all of these.  Later on, if we want to merge
-            some of these into other existing enocunters for the same patient on the same day, we can do so,
-            but that isn't needed for migration and might not be 100% accurate.
 
-            This could be something like a SQL script that runs the following statements:
+            /* Create encounters */
+            INSERT INTO `encounter`(`encounter_type`,`patient_id`,`encounter_datetime`,`creator`,`date_created`,`voided`,`voided_by`,`date_voided`,`void_reason`,`uuid`)
+            SELECT @encTId,patient_id,start_date,creator,date_created,voided,voided_by,date_voided,void_reason,UUID() FROM orders WHERE encounter_id is null;
+
+            /* Update orders*/
+            UPDATE orders O
+            LEFT JOIN encounter ENC ON O.patient_id = ENC.patient_id AND O.start_date=ENC.encounter_datetime AND O.creator=ENC.creator AND O.date_created=ENC.date_created
+            SET O.encounter_id=ENC.encounter_id
+            WHERE O.encounter_id is null AND ENC.encounter_type=@encTId;
+
```

Comment:
This seems like it will still work correctly with the changes I propose above.  Do you agree?  Can you clean up the formatting a bit please?

Reply:
Yes, It will work.